src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -3878,6 +3885,69 @@ public override IOperation VisitTypeOf(ITypeOfOperation operation, int? captureI
+        {
+            var parameterRef = new ParameterReferenceExpression(operation.Parameter, semanticModel: null,
+                operation.Syntax, operation.Parameter.Type, constantValue: default, isImplicit: true);
+            VisitInitializer(rewrittenTarget: parameterRef, initializer: operation);
+            return null;
+        }
+
+        public override IOperation VisitFieldInitializer(IFieldInitializerOperation operation, int? captureIdForResult)
+        {
+            foreach (IFieldSymbol fieldSymbol in operation.InitializedFields)
+            {
+                IInstanceReferenceOperation instance = fieldSymbol.IsStatic ?
+                    null :
+                    new InstanceReferenceExpression(InstanceReferenceKind.ContainingTypeInstance, semanticModel: null,
+                        operation.Syntax, fieldSymbol.ContainingType, constantValue: default, isImplicit: true);
+                Optional<object> constantValue = fieldSymbol.HasConstantValue ? new Optional<object>(fieldSymbol.ConstantValue) : default;
+                var fieldRef = new FieldReferenceExpression(fieldSymbol, isDeclaration: false, instance, semanticModel: null,
+                    operation.Syntax, fieldSymbol.Type, constantValue, isImplicit: true);
+                VisitInitializer(rewrittenTarget: fieldRef, initializer: operation);
+            }
+
+            return null;
+        }
+
+        public override IOperation VisitPropertyInitializer(IPropertyInitializerOperation operation, int? captureIdForResult)
+        {
+            foreach (IPropertySymbol propertySymbol in operation.InitializedProperties)
+            {
+                var instance = propertySymbol.IsStatic ?
+                    null :
+                    new InstanceReferenceExpression(InstanceReferenceKind.ContainingTypeInstance, semanticModel: null,
+                        operation.Syntax, propertySymbol.ContainingType, constantValue: default, isImplicit: true);
+                var propertyRef = new PropertyReferenceExpression(propertySymbol, instance, arguments: ImmutableArray<IArgumentOperation>.Empty,
+                    semanticModel: null, operation.Syntax, operation.Type, constantValue: default, isImplicit: true);
+                VisitInitializer(rewrittenTarget: propertyRef, initializer: operation);
+            }
+
+            return null;
+        }
+
+        private void VisitInitializer(IOperation rewrittenTarget, ISymbolInitializerOperation initializer)
+        {
+            bool haveLocals = !initializer.Locals.IsEmpty;
+            if (haveLocals)
+            {
+                EnterRegion(new RegionBuilder(ControlFlowGraph.RegionKind.Locals, locals: initializer.Locals));
+            }
+
+            // We skip constants in the control flow graph, as they're not actually involved in any control flow.
+            if (!rewrittenTarget.ConstantValue.HasValue && initializer.Value != null)
```

Comment:
>initializer.Value != null [](start = 59, length = 25)

Is this condition ever false? Then I would say we shouldn't even call this function in this case and not even allocate the target node.

Reply:
I wasn't able to come up with a unit test where this is ever false, but let if as a defensive check.

---
In reply to: [187455833](https://github.com/dotnet/roslyn/pull/26763#discussion_r187455833) [](ancestors = 187455833)

Reply:
> I wasn't able to come up with a unit test where this is ever false, but let if as a defensive check.

According to TetsOperationVisitor.VisitSymbolInitializer, Value is never null. 


---
In reply to: [187457335](https://github.com/dotnet/roslyn/pull/26763#discussion_r187457335) [](ancestors = 187457335,187455833)

Reply:
Let me replace the check with an assert.

---
In reply to: [187458019](https://github.com/dotnet/roslyn/pull/26763#discussion_r187458019) [](ancestors = 187458019,187457335,187455833)